tests/components/garmin_connect/test_config_flow.py

```diff
@@ -0,0 +1,122 @@
+"""Test the Garmin Connect config flow."""
+from unittest.mock import patch
+
+from garminconnect import (
+    GarminConnectAuthenticationError,
+    GarminConnectConnectionError,
+    GarminConnectTooManyRequestsError,
+)
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.garmin_connect import config_flow
+from homeassistant.components.garmin_connect.const import DOMAIN
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+MOCK_CONF = {
+    CONF_ID: "First Lastname",
+    CONF_USERNAME: "my@email.address",
+    CONF_PASSWORD: "mypassw0rd",
+}
+
+
+@pytest.fixture(name="mock_garmin_connect")
+def mock_garminconnect():
+    """Mock Garmin."""
+    with patch("homeassistant.components.garmin_connect.config_flow.Garmin") as garmin:
+        garmin().get_full_name.return_value = "First Lastname"
+        yield garmin
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.GarminConnectConfigFlowHandler()
+    flow.hass = hass
+    return flow
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_step_user(hass, mock_garmin_connect):
+    """Test registering an integration and finishing flow works."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user(user_input=MOCK_CONF)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "First Lastname"
+    assert result["data"] == MOCK_CONF
+
+
+async def test_connection_error(hass):
+    """Test for connection error."""
+    flow = init_config_flow(hass)
+
+    with patch(
+        "homeassistant.components.garmin_connect.config_flow.Garmin",
```

Comment:
Is it the instantiation of this class that will raise or is it when the method of the class instance is called? Ie, does the real class instantiation have real side effects?

If it's a method that should raise we could simplify and use the mocked instance that the fixture yields and just set a side_effect on the corresponding instance method.

Reply:
I used:
`"homeassistant.components.garmin_connect.config_flow.Garmin",`
because it mocked two methods at ones. I couldn't get them to get mocked directly like I think you prefer. (not using *config_flow* but garmin directly)
this is the code called from my configflowhandler:
```
client = garminconnect.Garmin(username, password)
client.get_full_name()
```
Cannot get to patch both instead of 
`"homeassistant.components.garmin_connect.config_flow.Garmin",`

Reply:
We would keep the fixture with the change I suggested above. This already patches the `Garmin` class. Then we can set return values or side effects on the class instance as needed here in the test. Ie, we don't need to patch again here.

Reply:
Eg:
```py
mock_garmin_connect.get_full_name.side_effect = Exception()
```

Reply:
Clear... working on that...thanks for the patience :smile: